pkg/parse/run.go

```diff
@@ -584,6 +592,48 @@ func (r *reconciler) startAsyncStatusUpdates(ctx context.Context) <-chan struct{
 	return doneCh
 }
 
+// unblockHydration adds the image digest into the ready-to-render file under the
+// shared directory between reconciler and hydration-controller to inform the
+// hydration-controller to proceed rendering
+func unblockHydration(commit string, r Reconciler) status.Error {
+	readyToRenderFile := r.Options().ReconcilerSignalsDir.Join(cmpath.RelativeSlash(hydrate.ReadyToRenderFile)).OSPath()
+	klog.Infof("writing commit %s to ready to render file %s", commit, readyToRenderFile)
+	// Overwrite the commit in ready-to-render file
+	file, err := os.OpenFile(readyToRenderFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0666)
+	if err != nil {
+		return status.OSWrap(err)
+	}
+	defer func() {
+		if closeErr := file.Close(); closeErr != nil {
+			klog.Errorf("failed to close file %s: %v", readyToRenderFile, closeErr)
+		}
+	}()
+
+	_, err = file.WriteString(commit)
+	if err != nil {
+		return status.OSWrap(err)
+	}
+	return nil
+}
+
+// blockHydration removes the ready-to-render file from the shared
+// directory and returns any error that occurs during the procedure except
+// "file not found". This function removes the ready-to-render file at its
+// best effort, since with source errors presenting, the ready-to-render
+// file will not be updated with the expected commit that is expected by
+// hydration-controller, thus rendering will be blocked.
+func blockHydration(r Reconciler) {
+	readyToRenderFile := r.Options().ReconcilerSignalsDir.Join(cmpath.RelativeSlash(hydrate.ReadyToRenderFile)).OSPath()
```

Comment:
This is repeated in both unblock and block functions. Can it be extracted to a variable?